### PR TITLE
Create a guide to using Teleport with Spacelift

### LINKED ISSUE #25421: Add a guide for using the Teleport Terraform provider with Spacelift
## Applies To
New guide in the section created in https://github.com/gravitational/teleport/issues/25418

## Details
Spacelift is a popular CD solution for managing infrastructure-as-code environments using GitOps. Let's create a guide to using Spacelift with our Terraform provider, which requires a bit of setup work (e.g., creating a Teleport user and issuing credentials) before you can use it. 

Ideally, users of other managed Terraform GitOps providers can also follow this guide as an example.

### PR DESCRIPTION
**Pull request by ptgott**
Closes #25421

### CODE REVIEW

File: docs/img/management/spacelift/apply-success.png
Comment (alexfornuto) [inline docs/img/management/spacelift/apply-success.png:1]: I suggest resizing all the new images, as they are 5k wide and gonna be big/slow loading no matter how compressed the are. They don't need to be any bigger than the maximum size our layout will scale to.

File: docs/pages/management/dynamic-resources/spacelift.mdx
Comment (alexfornuto) [inline docs/pages/management/dynamic-resources/spacelift.mdx:1]: @ptgott is this the same version as review/approved in #26031?

Comment (ptgott) [reply docs/pages/management/dynamic-resources/spacelift.mdx:1]: It is!

File: docs/img/management/spacelift/apply-success.png
Comment (ptgott) [reply docs/img/management/spacelift/apply-success.png:1]: I've resized the images and squashed my changes into the original commit. I've also added a sidebar entry for the Spacelift guide.

**Review by alexfornuto — APPROVED** [review]
Re-approving based on [original review](#discussion_r1261435213)

**Review by zmb3 — APPROVED** [review]

**Review by stevenGravy — APPROVED** [review]

**Pull request merged**